docs/containers.md

```diff
@@ -0,0 +1,39 @@
+# Containers & DB
+
+CourseTable uses the following Docker containers for core functionality, so they must be running for the app to work:
+
+- `express`: this contains the Express app code for running the API.
+- `mysql`: this is a MySQL database that stores user data.
+- `graphql-engine`: this is a Hasura engine that wraps the Postgres database created by Ferry. Ferry dumps course data in it, and the engine exposes a GraphQL API for querying the data.
+- `phpmyadmin`: this is a PHPMyAdmin instance that allows you to view the MySQL database managed by API. It is useful for debugging and DB manipulation.
+
+Note that we have two databases: a MySQL database managed by API, which stores user data, and a Postgres database managed by Ferry, which stores course data. The latter is exposed as a GraphQL API by the Hasura engine. Therefore, you need to be connected to the Internet to even start CourseTable locally, because the GraphQL engine used in dev still communicates with the remote Postgres database.
+
+In `coursetable/api`, we only manage the `express` and `graphql-engine` containers. We do provide development versions of the `mysql` and `phpmyadmin` containers, which should mirror the setup and table schema used in prod, but the actual prod configuration is located at [`coursetable/infra/mysql`](https://github.com/coursetable/infra/blob/main/mysql/docker-compose.yml).
+
+Here's the data flow for course data:
+
+1. Ferry writes course data to [`ferry-data`](https://github.com/coursetable/ferry-data) in JSON.
+2. Ferry then writes the data to a Postgres database on the prod server.
+3. The Postgres database exposes itself to the Internet.
+4. During development and prod, we spin up a Hasura engine that wraps the Postgres database and exposes a GraphQL API.
+5. The Express app queries the GraphQL API and generates static JSON again, located in the `api/static` folder. These are cached locally unless you run `./start.sh -d -o`.
+6. The frontend requests the Express endpoint, which serves these JSON files.
+
+And for user data:
+
+1. In dev, we always create a new MySQL database; in prod, we use the existing one. The table schema of both should be consistent with `api/mysql/database.sql`.
```

Comment:
@neilsong The dev DB \*uses\* `api/mysql/database.sql`. Maybe you mean `api/prisma/schema.prisma`? (All three should be synced anyway)

Reply:
Ah yea, table schema here is kind of an ambiguous term. The DB's true table schema is defined through executing database.sql but is then described in prisma language in schema.prisma. Either way, I agree that all three must be consistent - feel free to change that line however you see fit.